src/app/Library/CrudPanel/Traits/Query.php

```diff
@@ -219,25 +220,26 @@ private function getQueryColumnsFromWheres($query, $nested = false)
     /**
      * Get the query count without any filters or search applied.
      *
+     * @param Request $request
      * @return int
      */
-    public function getTotalEntryCount()
+    public function getTotalEntryCount(Request $request)
     {
         if (! $this->getOperationSetting('showEntryCount')) {
             return 0;
         }
-
-        return  $this->getRequest()->input('totalEntryCount') ??
+        
+        return  (int) ($request->request->get('totalEntryCount') ??
                 $this->getOperationSetting('totalEntryCount') ??
-                $this->getCurrentEntryCount();
+                $this->performQueryEntryCount());
     }
 
     /**
-     * Return the current query count.
+     * Runs the query count against the current crud panel query
      *
      * @return int
      */
-    public function getCurrentEntryCount()
+    public function performQueryEntryCount()
```

Comment:
I think this name is deceiving... 
If it returns something, it's not a `perform`, it's a `get`... It's a `function`, not a `method`.

I still think this function doesn't make sense to exist... let's just use `getEntryCount()` and that's it (`protected` as you made it here). Can't we do that?